library/std/src/thread/mod.rs

```diff
@@ -1759,6 +1759,18 @@ impl<T> JoinHandle<T> {
     /// }).unwrap();
     /// join_handle.join().expect("Couldn't join on the associated thread");
     /// ```
+    ///
+    /// # Notes
+    ///
+    /// This function has the same minimal guarantee regarding "foreign" unwinding operations (e.g.
+    /// an exception thrown from C++ code, or a `panic!` in Rust code compiled or linked with a
+    /// different runtime) as [`catch_unwind`]; namely, catching such an exception using this
+    /// function will have one of two behaviors, and it is unspecified which will occur:
+    ///
+    /// * The process aborts.
```

Comment:
The same goes here: it's good to specify whether the abort happens after unwinding out of the closure / to the thread entry, or upon joining the unwound thread (potentially arbitrarily later than the unwind, or even never).